docs/md/sink.md

```diff
@@ -0,0 +1,132 @@
+| `entityType`  | The entity type in the `workspace` to write outputs to.      |
+| `identifier`  | Selects the output that will be used as the entity name.     |
+| `fromOutputs` | Mapping from outputs to attribute names in the `entityType`. |
+
+#### `workspace`
+The workspace is a `"{workspace-namespace}/{workspace-name}"` string as it
+appears in the URL path in the Terra UI. The workspace must exist prior to
+workload creation. You must ensure that `workflow-launcher@firecloud.org` is
+a workspace "Writer" in order to write entities to the workspace.
+
+#### `entityType`
+The `entityType` is the name of the entity type in the workspace that entities
+will be created as. The entity type must exist prior to workload creation and
+must be a table in the workspace.
+
+#### `identifier`
+The `identifier` is the name of a pipeline output that should be used as the 
+name of each newly created entity. 
+
+Example - Let's say the pipeline you're running has an output called
+"sample_name" that uniquely identifies the inputs and outputs to that pipeline.
+By setting `"identifier": "sample_name"` in the sink configuration, entities
+will be created using the "sample_name" as the entity name.
+
+Note that when two sets of pipeline outputs share the same "identifier" value,
+the first set of outputs will be overwritten by the second in the workspace.
+
+#### `fromOutputs`
+`fromOutputs` configures how to create new entities from pipeline outputs by 
+mapping the output names to attributes in the `entityType`. Note that all 
+attribute names must exist in the entityType before the workload  creation.
+
+`fromOutputs` allows a small about of flexibility in how to construct an entity
+and supports the following relations:
+
+- `"attribute": "output"`
+  Direct mapping from an output to an attribute
+
+- `"attribute": ["output0", "output2"]`
+  Make an attribute from a list of pipeline outputs.
+
+- `"attribute": "$value"`
+  Use the literal "value" for an attribute.
+
+## Developer Guide
+A sink is one satisfying the `Sink` protocol as below:
+```clojure
+(defprotocol Sink
+  (update-sink!
+    ^Sink
+    [^Queue upstream ;; The queue to sink outputs from
+     ^Sink  sink     ;; This sink instance
+    ]
+    "Update the internal state of the `sink`, consuming objects from the
+     upstream `Queue`, performing any external effects as required.
+     Notes:
+     - The `Sink` and `Queue` are parameterised types and the `Queue`'s 
+       parameterisation must be convertible to the `Sink`s.
+     - Where possible, `Sink` implementations should minimise blocking and use
+       an internal queue to schedule and manage long-running computations."))
+```
+
+!!! info
+    The `Sink` protocol is currently implemented by the `update-sink!`
+    multimethod.
+
+To be used in a workload, a `Sink` implementation should satisfy the processing
+`stage` protocol and the `to-edn` multimethod in addition to the following
```

Comment:
What's the `stage` protocol?

Reply:
`validate` and `done?`

Reply:
You moved them into the `stage` namespace.

Reply:
Right.  Was confused since that isn't a protocol (yet!)